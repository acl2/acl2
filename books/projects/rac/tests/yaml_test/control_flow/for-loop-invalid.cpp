#include "ac_int.h"

// RAC begin

typedef ac_int<8, false> u8;

int adder(u8 a, u8 b) {
  int res = 0;
  for (res = b; a != 0; a--)
    res++;

  return res;
}

// RAC end
