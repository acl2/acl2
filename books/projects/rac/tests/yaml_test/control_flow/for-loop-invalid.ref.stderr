for-loop-invalid.cpp:9 (16-22): The variable (res) used in the recursive calls in never tested
  9 |   for (res = b; a != 0; a--)
    |                 ^^^^^^
 10 |     res++;
